modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala

```diff
@@ -727,10 +739,10 @@ object SharedOptions {
 
   // TODO: remove this state after resolving https://github.com/VirtusLab/scala-cli/issues/2658
   private val loggedDeprecatedToolkitWarning: AtomicBoolean = AtomicBoolean(false)
-  private def resolveToolkitDependency(
+  private def resolveToolkitDependencyAndScalaNativeVersionReqs(
     toolkitVersion: Option[String],
     logger: Logger
-  ): Seq[Positioned[AnyDependency]] = {
+  ): (Seq[Positioned[AnyDependency]], Seq[(String, String)]) = {
```

Comment:
Might worth to make a case class here

Reply:
putting a case class here would force me to handle it in `BuildOptions` macros, or else transform it back and forth into a tuple... I'll try to refactor it in a separate PR later